test/1_stdlib/StringAPI.swift

```diff
@@ -24,34 +24,44 @@ struct ComparisonTest {
   init(
     _ expectedUnicodeCollation: ExpectedComparisonResult,
     _ lhs: String, _ rhs: String,
-    xfail: TestRunPredicate = .custom({false}, reason: ""),
+    xfail: TestRunPredicate = .never,
     file: String = #file, line: UInt = #line
   ) {
+    self = ComparisonTest(expectedUnicodeCollation, lhs: lhs, rhs: rhs,
+      xfail: xfail, loc: SourceLoc(file, line, comment: "test data"))
+  }
+
+  init(
+    _ expectedUnicodeCollation: ExpectedComparisonResult,
+    lhs: String, rhs: String,
+    xfail: TestRunPredicate, loc: SourceLoc
+  ) {
     self.expectedUnicodeCollation = expectedUnicodeCollation
     self.lhs = lhs
     self.rhs = rhs
-    self.loc = SourceLoc(file, line, comment: "test data")
+    self.loc = loc
     self.xfail = xfail
   }
+
+  func replacingPredicate(_ xfail: TestRunPredicate) -> ComparisonTest {
+    return ComparisonTest(expectedUnicodeCollation, lhs: lhs, rhs: rhs,
+      xfail: xfail, loc: loc)
+  }
 }
 
-let comparisonTests = [
+let tests = [
```

Comment:
There definitely needs to be a little bit of documentation on both of the maps and on the special cases of the literal here explaining just that.  It's hard to see where and why things XFail without it.